src/main/scala/pirate/Prefs.scala

```diff
@@ -1,4 +1,31 @@
 package pirate
 
-sealed trait Prefs
-case object NullPrefs extends Prefs
+/**
+ * Prefs provides configuration options for generating
+ * a usage string as well as parser behaviour.
+ */
+case class Prefs(
+  backtrack: Boolean,
+  separateTopLevels: Boolean,
+  condenseSynopsis: Boolean,
+  usageOnError: Boolean,
+  flagIndent: Int,
+  descIndent: Int,
+  width: Int
+)
+
+/**
+ * Default preferences
+ *  - Explicit synopsis.
+ *  - 8/16 indents
+ *  - 80 width
+ */
+object NullPrefs extends Prefs(
```

Comment:
Just a note - this looks a little funny. You don't normally extend case classes (I actually thought you couldn't any more). Normally it would just be an object function.